doc/boot_protocol.rst

```diff
@@ -0,0 +1,274 @@
+only two boot protocols are supported, the Linux boot protocol and the
+Multiboot2 protocol.
+
+Linux Boot Tag
+^^^^^^^^^^^^^^
+
+The SKL relies on the DRTM Preamble to setup and stage the Linux kernel for
+booting, therefore the SKL only requires to be provided the location of the
+kernel's zero page to enable measuring and starting the Linux kernel.
+::
+
+    struct skl_tag_boot_linux {
+        struct skl_tag_hdr hdr;
+        u32 zero_page;
+    } __packed;
+
+Multiboot2 Boot Tag
+^^^^^^^^^^^^^^^^^^^
+
+The SKL relies on the DRTM Preamble to setup the Multiboot2 module list and
+must be provided the address for the Multiboot2 Boot Information (MBI) module
+and may optionally include the entry address and size of the kernel to be
+started.
+::
+
+    struct lz_tag_boot_mb2 {
+        struct skl_tag_hdr hdr;
+        u32 mbi;        
+        u32 kernel_entry;
+        u32 kernel_size;
+    } __packed;
+
+Linux Class
+-----------
+
+This class is for any boot information created/generated by the SKL that needs
+to be passed along to the Linux kernel.
+
+Setup Data Tag
+^^^^^^^^^^^^^^
+
+The Linux Secure Launch protocol requires a Setup Indirect setup data entry to
+pass information to Secure Launch. This resulting structure is chained into
+Linux's `struct setup_header->setup_data` linked list.
+::
+
+    struct skl_tag_setup_data {
+        struct skl_tag_hdr hdr;
+        struct setup_data data;
+    } __packed;
+
+Logging Class
+-------------
+
+The logging class holds the tags related to the TPM Event log and SKL's
+internal logging facilities.
+
+Event Log Tag
+^^^^^^^^^^^^^
+
+The Event Log tag provides three controls, policy, scheme, and minimum size.
+The policy controls how and what algorithms will be used, scheme allows for
+selecting the PCR usage scheme, and minimum size sets the minimum amount of Log
+Buffer space that will be used for the Event Log, see Log Buffer section for
+more details.
+::
+
+    struct skl_tag_evtlog {  
+        struct skl_tag_hdr hdr;
+        u16 policy;
+        u16 scheme;
+        u32 min_size;       
+    } __packed;             
+
+CRTM Hash Tag
+^^^^^^^^^^^^^
+
+The SKINIT instruction does not provide the measurement it made of the SKL but
+the value needs to be present in the Event Log for attestation purposes. This
+tag provides the option for the DRTM Preamble to hash the SKL and provide the
+measurement to the SKL for recording into the log, see CRTM Log Event for more
+details.
+::
+
+    struct skl_tag_crtm_hash {
+        struct skl_tag_hdr hdr;
+        u16 algo_id;
+        u8 digest[];
+    } __packed;
+
+Kernel Log Tag
+^^^^^^^^^^^^^^
+
+The Kernel Log tag provides a means to control/configure logging done by SKL
+itself. Specifically to enable/disable and set priority of the logging to be
+done by SKL.
+::
+
+    struct skl_tag_crtm_hash {
+        struct skl_tag_hdr hdr;
+        u8 enable;
+        u8 priority;
+    } __packed;
```

Comment:
You already mentioned it somewhere, but the location of log buffer is missing - just a reminder.

Reply:
Also, copy-paste of struct name from the previous tag.